script.js

```diff
@@ -120,55 +134,72 @@ const handleClick = (e) => {
       };
 
       
-      // shows all todos
-const showAllTodos = () => {
-  const todoItems = getTodoItems();
-  todoItems.forEach((item) => {
-    item.style.display = "block";
-  });
-};
-
-// shows todos without class completed
-const showActiveTodos = () => {
-  const todoItems = getTodoItems();
-  todoItems.forEach((item) => {
-    if (!item.classList.contains("completed")) {
-      item.style.display = "block";
-    } else {
-      item.style.display = "none";
-    }
-  });
-};
-
-// shows todos with class completed
-const showCompletedTodos = () => {
-  const todoItems = getTodoItems();
-  todoItems.forEach((item) => {
-    if (item.classList.contains("completed")) {
-      item.style.display = "block";
-    } else {
-      item.style.display = "none";
-    }
-  });
-};
-
-// delete all completed todos
-const deleteCompletedTodos = () => {
-  const todoItems = getTodoItems();
-  
-  todoItems.forEach((item) => {
-    if (item.classList.contains("completed")) {
-      item.remove();
-    } else {
-      item.style.display = "block";
-    }
-  });
-};
+      const applyToTodos = (condition, action) => {
+        const todoItems = ulList.querySelectorAll("li[data-id]");
+        todoItems.forEach((item) => {
+          if (condition(item)) {
+            action(item);
+          }
+        });
+      };
+      
+      const showAllTodos = () => {
+        applyToTodos(() => true, (listItem) => {
+          if (listItem) {
+            listItem.style.display = "block";
+          }
+        });
+      };
+      
+      const showActiveTodos = () => {
+        applyToTodos((item) => !item.classList.contains("completed"), (listItem) => {
+          if (listItem) {
+            listItem.style.display = "block";
+          }
+        });
+        applyToTodos((item) => item.classList.contains("completed"), (listItem) => {
+          if (listItem) {
+            listItem.style.display = "none";
+          }
+        });
+      };
+      
+      const showCompletedTodos = () => {
+        applyToTodos((item) => item.classList.contains("completed"), (listItem) => {
+          if (listItem) {
+            listItem.style.display = "block";
+          }
+        });
+        applyToTodos((item) => !item.classList.contains("completed"), (listItem) => {
+          if (listItem) {
+            listItem.style.display = "none";
+          }
+        });
+      };
+      
+      const deleteCompletedTodos = () => {
+        applyToTodos((item) => item.classList.contains("completed"), (listItem) => {
+          if (listItem) {
+            listItem.remove();
+          }
+        });
+      };
```

Comment:
Masz 4x użyte te same funkcje. Zoptymalizuj to.

 applyToTodos((item) => !item.classList.contains("completed"), (listItem) => {
          if (listItem) {
            listItem.style.display = "none";
          }
        });